doc/authoring_command_modules/authoring_commands.md

```diff
@@ -698,3 +700,27 @@ with self.argument_context('mymod') as c:
 ```
 
 Above are two examples of how this can be used. In the first instance, an Enum model is reflected from the SDK. In the second instance, a custom choice list is provided. This is preferable to using the native `argparse.choices` kwarg because the choice lists generated by `get_enum_type` will be case insensitive.
+
+## Extension Blocking
+
+It is possible for a command module to block specific extensions from being loaded.
+
+This is useful for commands that were once extensions that have now moved inside a command module.
+
+**Below is a recommended pattern for blocking specific extensions:**
+
+Here, we block an extension by name and also by version.
+
+This will allow the extension to be published in the future with the same name and a newer version that will not be blocked.
+
+This is great for experimental extensions that periodically get incorporated into the product.
+
+```Python
+class MyCommandsLoader(AzCommandsLoader):
+
+    def module_block_extension(self, ext):
+        """ Return True for the extension that should be blocked. An optional reason can be provided. """
+        from pkg_resources import parse_version
+        if ext.name == 'myextension' and ext.version and parse_version(ext.version) <= parse_version('0.5.0'):
+            return True, "The commands for myextension have moved inside the CLI. Consider removing the extension."
```

Comment:
I'm a little confused here. Is the guidance that the module author should add this method definition to their commands loader?  It looks like the infrastructure is plumbed through the CLI already.

What I would like to see from a UX standpoint is something like:
```Python
class MyCommandsLoader(AzCommandsLoader):
    def __init__(self, cli_ctx=None):
        blocked_extensions = [
          'azext_mycustomext1', # blanket blocks the extension by name
          'azext_mycustomext2 <= 0.5.0' # blocks by name/version combo
        ]
        super(ComputeCommandsLoader, self).__init__(cli_ctx=cli_ctx,
                                                    suppresses_extensions=blocked_extensions
                                                    )

```

In other words, let the module support another kwarg (add to the kwarg reference in the authoring doc) and the infrastructure just handles it, the same way it handles not loading any commands or params if the module isn't supported by the user's current profile.

Reply:
The only thing required by a command module is the following lines of code:
```Python
    def module_block_extension(self, ext):
        if ext.name == 'myextension':
            return True
```

The advantage of a callable that returns True if the extension should be blocked is:
1. I don't have to right custom parsing logic to handle strings like `'azext_mycustomext2 <= 0.5.0'` and determine the semantic meaning of it. I didn't want to do this.
2. An optional 'reason' can be specified and added to the result so command authors can state why they are suppressing an extension.
3. While I didn't include it in my example, the extension object that gets passed to `module_block_extension()` has all the metadata of that extension. So you can suppress by more than just the name and version. e.g. You can filter by say [author](https://github.com/Azure/azure-cli-extensions/blob/master/src/index.json#L63) if you only want to block it if it was the extension authored by you.

Reply:
I can probably change my usage of 'block' to 'suppress' like in the warning message and method names.

Reply:
I prefer suppress

Reply:
suppress seems preferable to me